Adhoc backup (#879)
percona/pmm: markdown/adhoc-backup.md

---
slug: 'adhoc-backup'
---

> ## Adhoc Backup

Reviewer: ```suggestion
## Ad hoc Backup
```

> PMM is able to backup the monitored servers. 

Reviewer 2: ```suggestion
PMM can backup the monitored servers. 
```

> In this section we will show you how can you take Adhoc Backups from a service.

Reviewer: ```suggestion
In this section we will show you how can you take ad hoc Backups from a service.
```

> In this section we will show you how can you take ad hoc Backups from a service.

Reviewer 2: ```suggestion
This section describes taking ad hoc backups from a service.
```

> ### Creating a Backup

Reviewer 3: should we mention about limitations we have to run backup?
@artemgavrilov

Author: Yes, I was thinking about that but the backup solution is still under development so it will change..  we have to update this doc when the development of the backup process is already done. But let me know what do you think.

Reviewer 4: I don't think that duplicate information is a good idea. We can just provide link to docs where we have requirements: https://docs.percona.com/percona-monitoring-and-management/using/backup.html#mysql-backup-preconditions

Reviewer 3: Cool, can we include that to this doc?

> Here is an example Curl API call to create a backup:

Reviewer 2: ```suggestion
Here is an example of a Curl API call to create a backup:

```

> ```

Reviewer: ```suggestion
```bash
```

curl --insecure -X POST -H 'Authorization: Bearer XXXXX' \
     --request POST \
     --url https://127.0.0.1/v1/management/backup/Backups/Start \
     --header 'Accept: application/json' \
     --header 'Content-Type: application/json' \
     --data '
{
     "service_id": "/service_id/XXXXX",
     "location_id": "/location_id/XXXXX",
     "name": "Test Backup",
     "description": "Test Backup",
     "retry_interval": "60s",
     "retries": 1
}
'
```

> You need the authentication string which is described [here](ref:authentication).

Reviewer 5: or you can replace `-H 'Authorization: Bearer XXXXX'` by `-u "<user>:<password>"`

> You need the authentication string which is described [here](ref:authentication).

Reviewer 2: ```suggestion
You require an authentication string described [here](ref:authentication).
```

> You need the authentication string which is described [here](ref:authentication).

Reviewer 2: ```suggestion
You require an authentication string which is described [here](ref:authentication).
```

> Then you need the [service_id](ref:listservices) and [location_id](ref:listlocations).

Reviewer 2: ```suggestion
Also, you require the [service_id](ref:listservices) and [location_id](ref:listlocations).
```

> You can choose your own `name` and `description` for the backup. You can also configure `retry_interval` and `retries` if that is necessary. 

Reviewer 2: ```suggestion
You can choose a `name` and `description` for the backup. You can also configure `retry_interval` and `retries` if required. 
```
